eZ/Publish/Core/Repository/LocationService.php

```diff
@@ -350,6 +350,10 @@ public function createLocation(ContentInfo $contentInfo, LocationCreateStruct $l
         $content = $this->repository->getContentService()->loadContent($contentInfo->id);
         $parentLocation = $this->loadLocation($locationCreateStruct->parentLocationId);
 
+        if (!$this->repository->canUser('content', 'manage_locations', $contentInfo)) {
```

Comment:
`$contentInfo => $content->contentInfo` as we have already loaded it can use the one we trust as opposed the one provided by user. _(note: I'm perosnally not a fan of the whole "can't trust objects from user" thing, as it kind of beats the purpose of requiring objects as opposed to just id's in the signature to begin with. But someone dealing with this in the past thought it was needed)_.

Also, any reason why you did not provide the context (parent location) like done below?

Reply:
apart from what @andrerom says, why try to load `$content` and `$parentLocation` before the policy checks? afaik those are not used before this checks, so, we could save some calls...

Reply:
Off topic :)

Reply:
> $contentInfo => $content->contentInfo as we have already loaded it can use the one we trust as opposed the one provided by user.

That's right @andrerom, I've changed it. 

> Also, any reason why you did not provide the context (parent location) like done below?

Just an oversight, fixed it too.

> apart from what @andrerom says, why try to load $content and $parentLocation before the policy checks?

@crevillo these two variables are indeed used in both policy checks. Previously, they weren't (see the first @andrerom comment), but as I mentioned above it was my mistake.